src/app/wallets/send-on-chain.ts

```diff
@@ -397,6 +382,50 @@ const executePaymentViaOnChain = async ({
     })
     if (journal instanceof Error) return journal
 
+    const txHash = await onChainService.payToAddress({
+      address,
+      amount: amountToSend,
+      targetConfirmations,
+      description: `journal-${journal.journalId}`,
+    })
+    if (
+      txHash instanceof InsufficientOnChainFundsError ||
+      txHash instanceof CPFPAncestorLimitReachedError
+    ) {
+      const reverted = await ledgerService.revertOnChainPayment({
+        journalId: journal.journalId,
+      })
+      return reverted || txHash
+    }
+    if (txHash instanceof Error) {
```

Comment:
Why aren't we reverting for all errors? In lightning we revert for all send errors except `LnPaymentPendingError` I think.